File: day5/main.py
def part_1():
    data_map = {
        1: ["D", "M", "S", "Z", "R", "F", "W", "N"],
        2: ["W", "P", "Q", "G", "S"],
        3: ["W", "R", "V", "Q", "F", "N", "J", "C"],
        4: ["F", "Z", "P", "C", "G", "D", "L"],
        5: ["T", "P", "S"],
        6: ["H", "D", "F", "W", "R", "L"],
        7: ["Z", "N", "D", "C"],
        8: ["W", "N", "R", "F", "V", "S", "J", "Q"],
        9: ["R", "M", "S", "G", "Z", "W", "V"],
    }
    with open("input.txt", "r") as file:
        data = file.read()
    lines = data.split("\n")
    for line in lines:
        values = line.split(" ")
        moves = int(values[1])
        from_stack = int(values[3])
        to_stack = int(values[5])

        count = 0
        while count < moves:
            val = data_map[from_stack].pop()
            data_map[to_stack].append(val)
            count += 1

    output = []
    keys = data_map.keys()
    for key in keys:
        output.append(data_map[key].pop())
    print("".join(output))


def part_2():
    data_map = {
        1: ["D", "M", "S", "Z", "R", "F", "W", "N"],
        2: ["W", "P", "Q", "G", "S"],
        3: ["W", "R", "V", "Q", "F", "N", "J", "C"],
        4: ["F", "Z", "P", "C", "G", "D", "L"],
        5: ["T", "P", "S"],
        6: ["H", "D", "F", "W", "R", "L"],
        7: ["Z", "N", "D", "C"],
        8: ["W", "N", "R", "F", "V", "S", "J", "Q"],
        9: ["R", "M", "S", "G", "Z", "W", "V"],
    }
    with open("input.txt", "r") as file:
        data = file.read()
    lines = data.split("\n")
    for line in lines:
        values = line.split(" ")
        moves = int(values[1])
        from_stack = int(values[3])
        to_stack = int(values[5])

        new = data_map[from_stack][-moves:]
        data_map[from_stack] = data_map[from_stack][:-moves]
        data_map[to_stack].extend(new)

    output = []
    keys = data_map.keys()
    for key in keys:
        output.append(data_map[key].pop())
    print("".join(output))


def main():
    part_1()
    part_2()


if __name__ == "__main__":
    main()

File: day5/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func partOne() {

	m := map[int][]string{
		1: []string{"D", "M", "S", "Z", "R", "F", "W", "N"},
		2: []string{"W", "P", "Q", "G", "S"},
		3: []string{"W", "R", "V", "Q", "F", "N", "J", "C"},
		4: []string{"F", "Z", "P", "C", "G", "D", "L"},
		5: []string{"T", "P", "S"},
		6: []string{"H", "D", "F", "W", "R", "L"},
		7: []string{"Z", "N", "D", "C"},
		8: []string{"W", "N", "R", "F", "V", "S", "J", "Q"},
		9: []string{"R", "M", "S", "G", "Z", "W", "V"},
	}

	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error Opening The File", err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		vals := strings.Split(scanner.Text(), " ")
		moves, _ := strconv.Atoi(vals[1])
		fromStack, _ := strconv.Atoi(vals[3])
		// toStack, _ := strconv.Atoi(vals[5])

		s := m[fromStack][:-moves]

		fmt.Println(s)

		// move 3 from 5 to 3
		// fmt.Printf("%d\n", moves)
		// fmt.Printf("%d\n", fromStack)
		// fmt.Printf("%d\n", toStack)
	}

	fmt.Println(m[1])

}

func main() {
	arr1 := [6]int{10, 11, 12, 13, 14, 15}
	myslice := arr1[len(arr1)-2:]
	arr1 = arr1[0 : len(arr1)-2]

	fmt.Printf("myslice = %v\n", myslice)
	fmt.Printf("length = %d\n", len(myslice))
	fmt.Printf("capacity = %d\n", cap(myslice))
	// partOne()
}
